src/bun.js/api/BunObject.zig

```diff
@@ -4053,8 +4053,7 @@ pub const Timer = struct {
                 if (this.cancelled) {
                     _ = uv.uv_timer_stop(&this.timer);
                 }
-                // libuv runs on the same thread
-                return this.runFromJSThread();
+                this.event_loop.enqueueTaskConcurrent(this.concurrent_task.from(this, .manual_deinit));
```

Comment:
can we do enter/exit?
```zig
this.event_loop.enter();
defer this.event_loop.exit();
```
Because we're already on the thread we can free on

Reply:
Yeah actually added enter/exit before `callWithGlobalThis` we should always do this even on unix

Reply:
fixed on https://github.com/oven-sh/bun/pull/9651/commits/073e1ae165e83f1951f55e3d4c396be3d6a77ec2